public/js/chart.js

```diff
@@ -20,49 +20,51 @@ import Loader from './loader';
 /**
  * Use to make the API request to get the Lighthouse chart data for a PWA.
  */
-const CHART_BASE_URL = '/api/lighthouse/graph/';
+export default class Chart {
 
-export default class LighthouseChart {
-
-  constructor() {
-    this.chartElement = document.getElementById('chart');
+  constructor(config) {
+    this.chartElement = config.chartElement;
+    this.url = config.url;
     this.loader = new Loader(this.chartElement, 'dark-primary-background');
   }
 
   _loadChartsApi() {
-    if (window.google && window.google.charts) {
-      console.log('Googler Charts Loader already loaded');      
-      return Promise.resolve(window.google);
-    }
-
-    console.log('Loading Googler Charts Loader');
     return new Promise((resolve, reject) => {
-      const script = document.createElement('script');
-      script.defer = true;
-      script.src = 'https://www.gstatic.com/charts/loader.js';
-      script.onload = () => {
-        resolve(window.google);
+      const chartScript = document.getElementById('google-chart');
+      if (chartScript) {
+        console.log('Googler Charts Loader already loaded');
+        if (window.google) {
+          resolve();
+        } else {
+          chartScript.addEventListener('load', resolve);
+        }
+      } else {
+        console.log('Loading Googler Charts Loader');
+        const script = document.createElement('script');
+        script.id = 'google-chart';
+        script.defer = true;
+        script.src = 'https://www.gstatic.com/charts/loader.js';
+        script.onload = resolve;
+        script.onerror = reject;
+        document.head.appendChild(script);
       }
-      script.onerror = reject;
-      document.head.appendChild(script);
     });
   }
+
   load() {
     this.loader.show();
-    this._loadChartsApi()
-      .then(google => {
```

Comment:
We were using resolve(window.google) on the _loadChartsApi promise previously. The reason was that by doing this we could avoid accessing the global on the window object from here. I know you've been through some issues making this work, but wondering if we could keep that.